src/app/inventory/store/stats.ts

```diff
@@ -389,24 +419,24 @@ function enhanceStatsWithPlugs(
   // stats in first.
   const sortedSockets = _.sortBy(createdItem.sockets.allSockets, (s) => s.plugOptions.length);
   for (const socket of sortedSockets) {
-    buildPlugStats(socket, statsByHash, statDisplays);
+    attachPlugStats(socket, existingStatsByHash, statDisplays);
   }
 }
 
 /**
- * Attaches a stats attribute to each plug in the socket which is an object of statTypeHash to the value it
- * will contribute to the items stats when socketed.
+ * Generates the stat modification map for each DimPlug in a DimSocket
+ * and attaches it to the DimPlug's stats property
  */
-function buildPlugStats(
+function attachPlugStats(
   socket: DimSocket,
-  statsByHash: { [statHash: number]: DimStat },
-  statDisplays: { [statHash: number]: DestinyStatDisplayDefinition }
+  statsByHash: NodeJS.Dict<DimStat>,
```

Comment:
I think I actually don't love these types - it misses the built-in documentation of what the key type is and what it's for. The old type is more verbose but it's clearly indexed on statHash.

Reply:
i'll swap it back over to named keys, but the NodeJS.Dict was to make the keys optional so later lookups are properly `| undefined`